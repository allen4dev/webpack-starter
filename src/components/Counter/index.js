import React, { useState } from 'react';
import { number } from 'prop-types';
import styled from '@emotion/styled';

import logo from 'images/logo.png';

const Wrapper = styled.div``;
const Total = styled.h1`
  color: hotpink;
`;

function Counter({ initialCount }) {
  const [count, setCount] = useState(initialCount);

  const increment = () => setCount(prevCount => prevCount + 1);
  const decrement = () => setCount(prevCount => prevCount - 1);

  return (
    <Wrapper>
      <Total data-testid="total">Count: {count}</Total>

      <button type="button" onClick={increment}>
        Increment
      </button>

      <button type="button" onClick={decrement}>
        Decrement
      </button>

      <figure>
        <img src={logo} alt="League of legends" />
      </figure>
    </Wrapper>
  );
}

Counter.defaultProps = {
  initialCount: 0,
};

Counter.propTypes = {
  initialCount: number,
};

export default Counter;
