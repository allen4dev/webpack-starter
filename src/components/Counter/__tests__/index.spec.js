import 'jest-dom/extend-expect';
import 'react-testing-library/cleanup-after-each';

import React from 'react';
import { render, fireEvent } from 'react-testing-library';

import Counter from '../index';

it('should render with defaults', () => {
  const { getByTestId } = render(<Counter />);

  expect(getByTestId('total')).toHaveTextContent('Count: 0');
});

it('should render with a initial count', () => {
  const { getByTestId } = render(<Counter initialCount={2} />);

  expect(getByTestId('total')).toHaveTextContent('Count: 2');
});

it('total should add 1 to the total after clicking on the increment button', () => {
  const { getByText, getByTestId } = render(<Counter />);

  const total = getByTestId('total');

  expect(total).toHaveTextContent('Count: 0');

  fireEvent.click(getByText(/increment/i));

  expect(total).toHaveTextContent('Count: 1');
});

it('total should subtract 1 to the total after clicking on the decrement button', () => {
  const { getByText, getByTestId } = render(<Counter initialCount={2} />);

  const total = getByTestId('total');

  expect(total).toHaveTextContent('Count: 2');

  fireEvent.click(getByText(/decrement/i));

  expect(total).toHaveTextContent('Count: 1');
});
