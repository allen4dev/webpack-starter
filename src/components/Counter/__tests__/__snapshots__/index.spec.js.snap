// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with defaults 1`] = `
.emotion-0 {
  color: hotpink;
}

<div
  class="emotion-2 emotion-3"
>
  <h1
    class="emotion-0 emotion-1"
    data-testid="total"
  >
    Count: 
    0
  </h1>
  <button
    type="button"
  >
    Increment
  </button>
  <button
    type="button"
  >
    Decrement
  </button>
  <figure>
    <img
      alt="League of legends"
      src="[object Object]"
    />
  </figure>
</div>
`;
